### ISSUE TITLE
Dark Reader mode

### ISSUE THREAD

**Issue by jornmo** [QupZilla/qupzilla#1891]
I've recently come across [Dark Reader](https://chrome.google.com/webstore/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh) for Chrome, and I must say this is a really good feature that would have been awesome in Qupzilla!

>  it uses special intellectual way to transform bright blinding stuff into eye relaxing, high contrast, deluminated, dark themed and just smart looking web-pages which are easy to read at night. Use this extension to take care of your eyes.

![Dark Reader](https://lh3.googleusercontent.com/jOfSX_siDCgGKDG5bONY9SvbaHNQcgjBumPZ_IVVwI-PQQgr6ivLGXdIVSj2ZtYpPTlYXczD=s640-h400-e365-rw)

**Comment by jamesqf** [comment]
It's possible to do this for most things with a user stylesheet.  There are a few things I haven't figured out how to take care of yet - Google Maps being the most notable - but it's fairly easy to switch.

I do wish qupzilla had a way to keep a list of frequently-used stylesheets, though, so I wouldn't have to type the name each time.

**Comment by Dax89** [comment]
If can be useful, I have created this user defined css for my mobile browser, feel free to use it: https://github.com/Dax89/harbour-webpirate/blob/master/harbour-webpirate/qml/js/helpers/NightMode.js

It contains some extra code, but it can be adapted easily! :)

**Comment by jornmo** [comment]
Stylesheets and inverting colours is a good start, but I think that algorithms to detect darker colours that should not be inverted would be a good addition. Take arstechnica for an example. Simply inverting it would not do any good as it is already black.

**Comment by jamesqf** [comment]
The arstechnica site seems to work ok with my CSS.  I didn't know about it already having a black background, since I've been using various versions of the CSS since... well, just about forever.  Pretty much everything shows up as white text on black, and in my preferred font.  

But the point of the CSS (as far as I understand it - I'm no expert) is that it doesn't use algorithms to detect colors &c, it DICTATES what will be used.  I don't really see how a user could incorporate algorithm-level function into qupzilla?  It'd even be helpful to vary the CSS per web site, so I could for instance not turn off background images in Google Maps, where they are the map (I think).

**Comment by Dax89** [comment]
Color detection and Blacklisting can be done with a user defined Javascript, the CSS should be applied from there too

**Comment by jamesqf** [comment]
How would you get qupzilla to use a user-defined Javascript?  I can select a user CSS from the Preferences->Other menu, but I haven't noticed any equivalent for Javascript.  (But then, my Javascript knowledge is pretty much nonexistent.)

**Comment by Dax89** [comment]
You can enable GreaseMonkey extension from settings and define a new userscript for there

**Comment by jornmo** [comment]
It would be really nice if there was a preconfigured button for this, and not that everyone has to "hack" with CSS and JS to get it to work :)

**Comment by GreenLunar** [comment]
This issue should be posted at https://github.com/QupZilla/qupzilla-plugins/issues

I am in favor of such feature as plugin.  I know many people, especially people who are working on offices, who are using Mozilla browsers just because of their Reader extension.

**Comment by AcarBurak** [comment]
Yes, really that chromium extension so far works and works beautifully contrary to all the other darkish userstyles I've tried which do not work and are ugly hacks. I wish this work was part of QupZilla itself. Here's its GitHub page:
https://github.com/alexanderby/darkreader
